### DISCUSSION TITLE
Improved type warnings

### DISCUSSION THREAD

**Discussion by mynomoto** [clj-kondo/clj-kondo#1234] [Ideas]
Hello,

I would like to improve type warnings. One thing that I miss and think that would be great is having type information from maps used elsewhere. Let me show a concrete example:
```clj
(ns foo)

(defn foo
  [x]
  (inc (:a x)))

; config:
{:linters
 {:type-mismatch
  {:level :error
   :namespaces '{foo {foo {:arities {1 {:args [{:op :keys
                                                :req {:a :string}}]
                                        :ret :int}}}}}}}}
```
In this case `(inc (:a x))` should be an error because `(:a x)` is a string as declared on the config.
Would that be something that makes sense on clj-kondo?

**Comment by borkdude** [comment]
I think that makes sense yes.
